src/Components/Web/src/Forms/InputFile.cs

```diff
@@ -82,9 +82,7 @@ protected override void BuildRenderTree(RenderTreeBuilder builder)
         }
 
         internal Stream OpenReadStream(BrowserFile file, long maxAllowedSize, CancellationToken cancellationToken)
-            => _jsUnmarshalledRuntime != null ?
-                new SharedBrowserFileStream(JSRuntime, _jsUnmarshalledRuntime, _inputFileElement, file) :
-                new RemoteBrowserFileStream(JSRuntime, _inputFileElement, file, Options.Value, maxAllowedSize, cancellationToken);
+            => new BrowserFileStream(JSRuntime, _inputFileElement, file, Options.Value, maxAllowedSize, cancellationToken);
```

Comment:
Technically, this could be a breaking change for WASM environments as previously we didn't enforce the `maxAllowedSize`. Given that this appears to be a bug, I'd classify this under "acceptable breaking changes to prevent future issues". Any objections?

Reply:
Interesting. That is a bit breaky. Are there particular future issues you're anticipating?

Technically we could - if we want - check if the OS is "browser" and change pass `long.MaxValue` or similar here in that case.

@javiercn @pranavkm Opinions?

Reply:
On one hand, it's weird the option does not get used. At the same time, it's a bit unclear what the benefit of enforcing a size limit in WASM is. I'm cool with using MaxValue and documenting that the maxAllowedSize is ignored in WASM.

Reply:
> unclear what the benefit of enforcing a size limit in WASM is.

One potential use case is the developer only wants to process files <10MB as anything larger would not provide an optimal user experience / wouldn't make sense in the application context. This is where such a limit may be useful.

We can definitely just ignore the limit via `long.MaxValue` as Steve and Pranav suggest, just wanted to get consensus on the approach we take.

Reply:
> One potential use case is the developer only wants to process files <10MB as anything larger would not provide an optimal user experience / wouldn't make sense in the application context.

I agree that's a valid scenario, but they wouldn't want to achieve that by getting an exception from us, would they? They can already check the size property on the `BrowserFile` instances. Then they don't have to take guesses about what sort of exception to catch (etc) to trigger their "sorry this is too big" UI.